project-1_test/main.py

```diff
@@ -0,0 +1,144 @@
+import inspect
+import logging
+import types
+
+import gspread
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.chrome.service import Service as ChromeService
+from webdriver_manager.chrome import ChromeDriverManager
+from selenium.common.exceptions import NoSuchElementException
+import os, json
+from dotenv import load_dotenv
+load_dotenv()
+KEY = os.getenv("PROJECT_1202")
+option = webdriver.ChromeOptions()
+option.add_argument('--headless')
+driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()), options=option)
+link = "https://www.lusha.com/company-search/accounting/10/canada/193/page/2/"
+driver.get(url=link)
+
+gc = gspread.service_account("/sadique_timileyin/project-1_test/my-project.json")
+
+
+def open_spreadsheet(sheet_name):
+    """
+    This opens a spreadsheet already created on Google sheet
+    """
+    if isinstance(sheet_name, str):
+        sh = gc.open(sheet_name)
+        logging.info("spreadsheet opened")
+        return sh
+    else:
+        raise ValueError("The sheet name expected a string object")
+
+
+def create_spreadsheet(sheet_name):
+    """
+    This creates a new spreadsheet file in google sheet
+    """
+    if isinstance(sheet_name, str):
+        sh = gc.create(sheet_name)
+        logging.info("Spreadsheet created")
+        return sh
+    else:
+        raise ValueError("The sheet name expected a string object")
+
+
+def select_spreadsheet(filename, s_name):
+    """
+    This function  selects the preferred worksheet
+    :param filename: This is the name of the spreadsheet
+    :param s_name: This is the name of the desired worksheet in str format
+    """
+    if isinstance(s_name, str) and isinstance(filename, str):
+        worksheets = open_spreadsheet(filename).worksheet(s_name)
+        logging.info(f"Worksheet {s_name} has been loaded")
+        return worksheets
+    else:
+        raise ValueError("The input expected a string object")
+
+
+def spreadsheet_format(filename, s_name, col1, col2):
+    """
+    This functon is for the formatting of the spreadsheet
+    :param s_name: Name of current worksheet
+    :param filename: Name of spreadsheet
+    :param col1: This is the name of the start column in the range in str format
+    :param col2: This is the name of the end column in the range in str format
+    """
+    if isinstance(col1, str) and isinstance(col2, str) and isinstance(filename, str) and isinstance(s_name, str):
```

Comment:
Find a way to reduce your conjuction here, we shouldn't be having so many logic operators in a single flow, since this is also checking if all are strings, there'd be a better way to rewrite this